src/vehicle_use.cpp

```diff
@@ -1908,55 +1908,36 @@ void vehicle::use_bike_rack( int part )
-    const bool has_towel = avail_part_with_feature( interact_part, "TOWEL", true ) >= 0;
-    const bool has_weldrig = avail_part_with_feature( interact_part, "WELDRIG", true ) >= 0;
-    const bool has_chemlab = avail_part_with_feature( interact_part, "CHEMLAB", true ) >= 0;
-    const bool has_purify = avail_part_with_feature( interact_part, "WATER_PURIFIER", true ) >= 0;
-    const bool has_controls = avail_part_with_feature( interact_part, "CONTROLS", true ) >= 0;
-    const bool has_electronics = avail_part_with_feature( interact_part, "CTRL_ELECTRONIC", true ) >= 0;
-    const int cargo_part = part_with_feature( interact_part, "CARGO", false );
-    const bool from_vehicle = cargo_part >= 0 && !get_items( cargo_part ).empty();
-    const bool can_be_folded = is_foldable();
-    const bool is_convertible = tags.count( "convertible" ) > 0;
-    const bool remotely_controlled = g->remoteveh() == this;
-    const int autoclave_part = avail_part_with_feature( interact_part, "AUTOCLAVE", true );
-    const bool has_autoclave = autoclave_part >= 0;
-    bool autoclave_on = ( autoclave_part == -1 ) ? false :
-                        parts[autoclave_part].enabled;
-    const int washing_machine_part = avail_part_with_feature( interact_part, "WASHING_MACHINE", true );
-    const bool has_washmachine = washing_machine_part >= 0;
-    bool washing_machine_on = ( washing_machine_part == -1 ) ? false :
-                              parts[washing_machine_part].enabled;
-    const int dishwasher_part = avail_part_with_feature( interact_part, "DISHWASHER", true );
-    const bool has_dishwasher = dishwasher_part >= 0;
-    bool dishwasher_on = ( dishwasher_part == -1 ) ? false :
-                         parts[dishwasher_part].enabled;
-    const int monster_capture_part = avail_part_with_feature( interact_part, "CAPTURE_MONSTER_VEH",
-                                     true );
-    const bool has_monster_capture = monster_capture_part >= 0;
-    const int bike_rack_part = avail_part_with_feature( interact_part, "BIKE_RACK_VEH", true );
-    const int harness_part = avail_part_with_feature( interact_part, "ANIMAL_CTRL", true );
-    const bool has_harness = harness_part >= 0;
-    const bool has_bike_rack = bike_rack_part >= 0;
-    const bool has_planter = avail_part_with_feature( interact_part, "PLANTER", true ) >= 0 ||
-                             avail_part_with_feature( interact_part, "ADVANCED_PLANTER", true ) >= 0;
-    const int workbench_part = avail_part_with_feature( interact_part, "WORKBENCH", true );
-    const bool has_workbench = workbench_part >= 0;
+    const bool has_items_on_ground = here.sees_some_items( vp.pos(), player_character );
+    const bool items_are_sealed = here.has_flag( "SEALED", vp.pos() );
+    const turret_data turret = turret_query( vp.pos() );
+    // next batch of autos are typed cata::optional<vpart_reference>
+    const auto curtain_part = vp.avail_part_with_feature( "CURTAIN" );
```

Comment:
these are all `int`s, right?  Because that's what `avail_part_with_feature()` returns.  So how about you actually save a character per line and simplify future code maintenance by specifying the exact type?

Reply:
Wasn't aware of your stance against autos I don't mind them either way - I'll fix it up soon

Reply:
Ooh I misunderstood what you're saying actually sorry, those are not ints - I've extended `vpart_position` with `avail_part_with_feature` so they're `cata::optional<vpart_reference>` s

Reply:
okay, well, I'm not as smart as a compiler, obviously.  But yes, `auto` is a keyword that we would generally like to see less used inside of CDDA.  If you're doing a type declaration and it's easy for you to figure out the type, then we would generally prefer that state the type explicitly.

Reply:
Not sure it's visible if i just push a commit - when you have time see the last commit - i got rid of autos